LOG4J2-2850 Fix ctor used in LocalizedMessageFactory#newMessage().

sandeepbarnwal opened the pull request:
newMessage method in LocalizedMessageFactory uses incorrect constructor. it creates the new message with baseName as messagePattern. I have changed it to point to the appropriate constructor.

vy commented:
Thanks so much @sandeepbarnwal! Would you mind updating `/src/changes/changes.xml` and adding a simple test (to `LocalizedMessageTest`?) as well, please?

sandeepbarnwal commented:
> Thanks so much @sandeepbarnwal! Would you mind updating `/src/changes/changes.xml` and adding a simple test (to `LocalizedMessageTest`?) as well, please?

Sure and thanks for correcting the typo in the description. Added the test case in LocalizedMessageFactoryTest as the change in that class.

> Thanks so much @sandeepbarnwal! Would you mind updating `/src/changes/changes.xml` and adding a simple test (to `LocalizedMessageTest`?) as well, please?

Sure. I have added the test cases and also thanks for the correction in the description.
